services/core/shared/core/delegates.js

```diff
@@ -119,8 +132,113 @@ const getTotalNumberOfDelegates = async (params = {}) => {
 	return relevantDelegates.length;
 };
 
+const computeDelegateRankAndStatus = async () => {
+	const db = await pouchdb(config.db.collections.delegates.name);
+	const sdkVersion = Number(coreApi.getSDKVersion().split('sdk_v')[1]);
+	const numActiveForgers = 101;
+
+	const delegateStatus = {
+		ACTIVE: 'active',
+		STANDBY: 'standby',
+		BANNED: 'banned',
+		PUNISHED: 'punished',
+		NON_ELIGIBLE: 'non-eligible',
+	};
+
+	const allDelegates = await db.findAll();
+	const lastestBlock = await getBlocks({ limit: 1 });
+	const allNextForgersAddressList = nextForgers.map(forger => forger.account.address);
```

Comment:
Is the `nextForgers` sorted by its voting power?

Reply:
Since `nextForgers` are also delegates, they're sorted based on the same logic as delegates:
1. Descending order of their `delegateWeight`
2. In case of ties in the previous step, buffer sorted based on their addresses